astropy/io/fits/hdu/hdulist.py

```diff
@@ -830,6 +841,10 @@ def _readfrom(cls, fileobj=None, data=None, mode=None,
 
         saved_compression_enabled = compressed.COMPRESSION_ENABLED
 
+        # (extname, extver) tuple used by convenience functions to only load as
+        # much as necessary
+        last_read_ext = kwargs.pop("_last_read_ext", None)
+        idx = 0
```

Comment:
Instead of defining the `idx` here you could also replace the `while`-loop with `for idx in itertools.count()`. This is just a suggestion, feel free to ignore it.